### ISSUE TITLE
Whatsapp Error (Needs Chrome 36+)

### ISSUE THREAD

**Issue by mikeashelby** [ramboxapp/community-edition#1981]
I downloaded Rambox today, but the WhatsApp service isn't working. It Says:

> WhatsApp works with Google Chrome 36+
> To use WhatsApp, update Chrome or use Mozilla Firefox, Safari, Microsoft Edge or Opera.
> UPDATE GOOGLE CHROME

Any ideas?

	
	-
	> Rambox 0.6.2
	> Electron 2.0.8
	> darwin x64 17.7.0

**Comment by valentine** [comment]
It seems like there's some sort of feature detection because a simple User Agent change isn't working.

**Comment by mahovich** [comment]
+1
[screenshot](https://i.imgur.com/q5UDnYA.png)

**Comment by stoque2** [comment]
I get the same issue. It worked once for me after launching it for the first time after upgrading from Ubuntu 16 to 18. It seems to have to do with an error in Whatsapp Web's browser detection.

**Comment by trinaldi** [comment]
[Same](https://i.imgur.com/UiJY05Y.png) with Arch Linux:

**`uname -a`:**
`Linux 4.19.4-arch1-1-ARCH #1 SMP PREEMPT Fri Nov 23 09:06:58 UTC 2018 x86_64 GNU/Linux`


**About Rambox:**
```
Version: 0.6.2
Platform: linux (x64)
Electron: 2.0.8
Chromium: 61.0.3163.100
Node: 8.9.3
```

**Comment by SBI-boy** [comment]
It seems that nobody is answering here, it might be that the devs are only focused in the paid version, it's a shame :/

**Comment by mikeashelby** [comment]
I noticed that the same is true on Franz, so I don't think it's a paid/free
thing - I would imagine it's happening on the paid version too. And also
could be an issue on WhatsApp's end with their browser detection rather
than rambox itself.

Still, somewhat frustrating to not be able to put all my messaging
platforms in one place!

That said, it's otherwise really awesome!

**Comment by pbek** [comment]
for some reason I have the same problem on Windows, but not on Linux (KDE Neon and Manjaro) or macOS (High Sierra)

**Comment by stoque2** [comment]
What I've figured out so far is that going into developer tools (right click on Whatsapp tab) and selecting the memory tab, then deleting all site data resolves the issue but only until the next start of rambox.

**Comment by trinaldi** [comment]
@stoque2 That's a temporary solution as mentioned [here](https://github.com/meetfranz/franz/issues/1138#issuecomment-443166406) (It's Franz repo, but the same error).

**Comment by pbek** [comment]
> but not on Linux (KDE Neon and Manjaro) or macOS (High Sierra)

I take that back, after an app restart I have the same troubles...

**Comment by simonszu** [comment]
> It seems that nobody is answering here, it might be that the devs are only focused in the paid version, it's a shame :/

Somewhere in here is a bug i have opened in 2017, with no reply so far except the autobot who tried to close it since there weren't any replies.

**Comment by trinaldi** [comment]
Franz seems to have fixed the issue.

**Comment by geek4teck** [comment]
Same issue on Windows 10 x64 & Ubuntu 16.04

Problem seems to be occuring 4 - 5 days back

**Comment by localhorst** [comment]
> Same issue on Windows 10 x64 
> Problem seems to be occuring 4 - 5 days back

I am on a fresh install of Windows 10.
Same problem, it started yesterday (Dec 12, 2018)  

Rambox Version: 0.6.2
Platform: win32 (x64)
Electron: 2.0.8
Chromium: 61.0.3163.100
Node: 8.9.3

**Comment by localhorst** [comment]
So here is a workaround using a custom service:

**Step 1**
Create Custom service

**Step 2**
Name: Whatever
URL: https://web.whatsapp.com/
Logo: https://upload.wikimedia.org/wikipedia/commons/thumb/6/6b/WhatsApp.svg/64px-WhatsApp.svg.png

**Step 3**
Advanced - Custom Code: 
```
function setUserAgent(window, userAgent) {
    // Works on Firefox, Chrome, Opera and IE9+
    if (navigator.__defineGetter__) {
        navigator.__defineGetter__('userAgent', function () {
            return userAgent;
        });
    } else if (Object.defineProperty) {
        Object.defineProperty(navigator, 'userAgent', {
            get: function () {
                return userAgent;
            }
        });
    }
    // Works on Safari
    if (window.navigator.userAgent !== userAgent) {
        var userAgentProp = {
            get: function () {
                return userAgent;
            }
        };
        try {
            Object.defineProperty(window.navigator, 'userAgent', userAgentProp);
        } catch (e) {
            window.navigator = Object.create(navigator, {
                userAgent: userAgentProp
            });
        }
    }
}

setUserAgent(window, "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/70.0.3538.77 Safari/537.36");
```

**Comment by mikeashelby** [comment]
Thanks for this! Brilliant!

**Comment by wgranicki** [comment]
@localhorst's workaround didn't work for me at first but after I've modified the last line, it works fine:
`setUserAgent(window, "Chrome/70.0.3538.77");`

Mind you, I have this exact version of Chrome installed.

**Comment by saenzramiro** [comment]
Overriding the User Agent seems like is not stable, because sometimes the warning comes again. We are currently investigating how to bypass this. Seems like WhatsApp want to prevent apps like us to use their service. 😭

@sudhagarc @andrewhassan Hey guys, any help with this, please? 🙏🏼

**Comment by mikeashelby** [comment]
Thanks Ramiro. I appreciate your time on the project!

**Comment by Laski** [comment]
This is a duplicate of #1979

**Comment by saenzramiro** [comment]
The pre-release v0.6.3 is available for those who want to test it.

**Comment by bjo81** [comment]
![bildschirmfoto_2018-12-05_21-55-05](https://user-images.githubusercontent.com/1674578/49543435-899a1100-f8d8-11e8-83b5-e88e00d64c7c.png)

Same issue with 0.6.3.

**Comment by trinaldi** [comment]
@bjo81 I just upgrade my binaries (Arch Linux, as mentioned [here](https://github.com/ramboxapp/community-edition/issues/1981#issuecomment-443429489)]) and the issue is fixed.

Did you try cleaning up the storage, uninstalling the service (in this case WhatsApp) and installing again or even a full uninstall of the app?

**Comment by wgranicki** [comment]
Can confirm 0.6.3 works for me (Win 10).
The problem seemed to persist after an upgrade of Rambox, but after I'd reloaded the service, it worked.
Closing and reopening Rambox doesn't bring the problem back.

**Comment by bjo81** [comment]
@trinaldi Now it works, thanks.

**Comment by saenzramiro** [comment]
Yes guys, need a force-reload or removing WhatsApp and adding it again.

**Comment by roccomuso** [comment]
so what is the final fix?

**Comment by yafp** [comment]
> so what is the final fix?

Removing and re-adding whatsapp helped for me (using 0.6.3)

**Comment by juliusHuelsmann** [comment]
I still have got the same issue on arch linux, fully updated with the Rambox 0.6.3 AppImage.

**Comment by driehle** [comment]
Updating to Rambox 0.6.3 and force-reloading solved the issue for me (Windows 10).
